### PR TITLE
Story/vspc 15

### PR DESCRIPTION
**Pull request by Neehaarika**
Added repository Interfaces and other changes

### CODE REVIEW

**Comment by Neehaarika** [comment]
Fixed Formatting

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/IBranchingPoint.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package edu.asu.diging.vspace.core.model;
+
+import java.util.List;
+
+public interface IBranchingPoint {
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/IBranchingPoint.java]: this needs to extend ISlide

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/BranchingPoint.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.hibernate.annotations.Parameter;
+
+import edu.asu.diging.vspace.core.model.IBranchingPoint;
+import edu.asu.diging.vspace.core.model.IChoice;
+
+@Entity
+public class BranchingPoint extends Slide implements IBranchingPoint {
+
+    @Id
+    @GeneratedValue(generator = "id-generator")
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/BranchingPoint.java]: this needs to be the same as the name of the genericgenerator defined below I believe

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Choice.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.hibernate.annotations.Parameter;
+
+import edu.asu.diging.vspace.core.model.IChoice;
+import edu.asu.diging.vspace.core.model.ISequence;
+
+@Entity
+public class Choice extends VSpaceElement implements IChoice {
+
+    @Id
+    @GeneratedValue(generator = "id-generator")
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Choice.java]: same as above

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Sequence.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import org.hibernate.annotations.Parameter;
+
+import edu.asu.diging.vspace.core.model.ISequence;
+import edu.asu.diging.vspace.core.model.ISlide;
+
+@Entity
+public class Sequence extends VSpaceElement implements ISequence{
+
+	@Id
+	@GeneratedValue(generator = "id-generator")
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Sequence.java]: same as above

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Sequence.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	
+	/* (non-Javadoc)
+	 * @see edu.asu.diging.vspace.core.model.impl.ISequence#setSlides(java.util.List)
+	 */
+	@Override
+	public void setSlides(List<ISlide> slides) {
+		this.slides = slides;
+	}
+	
+}
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Sequence.java]: indentation

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Slide.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import org.hibernate.annotations.Parameter;
+
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.IVSImage;
+
+@Entity
+public class Slide extends VSpaceElement implements ISlide {
+	
+	@Id
+	@GeneratedValue(generator = "id-generator")
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Slide.java]: same as above

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Slide.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	@GenericGenerator(name = "slide-id-generator", 	
+	  parameters = @Parameter(name = "prefix", value = "SLI"),
+	  strategy = "edu.asu.diging.vspace.core.data.IdGenerator")
+	private String id;
+	
+	@OneToOne(targetEntity=VSImage.class)
+	private IVSImage image;
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Slide.java]: let's make this a list. there could be more images on one slide.

**Comment by Neehaarika** [comment]
Please Review.

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Slide.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+public class Slide extends VSpaceElement implements ISlide {
+	
+	@Id
+	@GeneratedValue(generator = "slide-id-generator")
+	@GenericGenerator(name = "slide-id-generator", 	
+	  parameters = @Parameter(name = "prefix", value = "SLI"),
+	  strategy = "edu.asu.diging.vspace.core.data.IdGenerator")
+	private String id;
+	
+	@OneToOne(targetEntity=VSImage.class)
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Slide.java]: how can this be a onetoone if it's annotating a list?

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
Have you tried to run your code?
`Caused by: org.hibernate.AnnotationException: mappedBy reference an unknown target entity property: edu.asu.diging.vspace.core.model.impl.Sequence.module in edu.asu.diging.vspace.core.model.impl.Module.sequences
`

**Comment by jdamerow** [comment]
ok to test

File: vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Module.java [hunk 9]
```diff
@@ -1,36 +1,119 @@
...
+import edu.asu.diging.vspace.core.model.ISlide;
 
 @Entity
 public class Module extends VSpaceElement implements IModule {
 
-	@Id 
-	@GeneratedValue(generator = "id-generator")
+    @Id
+    @GeneratedValue(generator = "id-generator")
     @GenericGenerator(name = "id-generator",
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/core/model/impl/Module.java]: a generator with that id already exists

**Comment by jdamerow** [comment]
retest this please

**Pull request merged**